pyteal/ast/pragma_test.py

```diff
@@ -0,0 +1,150 @@
+import pytest
+import pkg_resources
+
+import pyteal as pt
+from pyteal.ast.pragma import __convert_pep440_compiler_version
+
+
+@pytest.fixture
+def mock_version(version: str, monkeypatch: pytest.MonkeyPatch):
+    def mocked_require(name: str):
+        if (
+            name == "pyteal"
+            and version is not None  # don't mock if no version is specified
+        ):
+            return [
+                pkg_resources.Distribution(
+                    version=version,
+                )
+            ]
+        else:
+            return pkg_resources.require(name)[0]
+
+    monkeypatch.setattr(pkg_resources, "require", mocked_require)
+
+
+@pytest.mark.parametrize(
+    "compiler_version,expected",
+    [
+        ("2", "2.0.0"),
+        (">=0.12.0a9.post2.dev9", ">=0.12.0-alpha9.2.9"),
+        ("<0.5.0 || >=1.0.0a9.post1.dev2", "<0.5.0 || >=1.0.0-alpha9.1.2"),
+        ("v0.12.9.post1 - v1.13.0.dev1", "0.12.9-1.0 - 1.13.0-0.1"),
+        (
+            "1.2.3a4.post5.dev6+AVM7.1",
+            "1.2.3-alpha4.5.6+avm7.1",
+        ),  # local versions are lowercased to be consistent with pkg_resources
+    ],
+)
+def test_convert_pep440_compiler_version(compiler_version, expected):
+    assert __convert_pep440_compiler_version(compiler_version) == expected
+
+
+@pytest.mark.usefixtures("mock_version")
+@pytest.mark.parametrize(
+    "version, compiler_version, should_error",
+    [
+        # valid
```

Comment:
Could you add a test with the carrot operator `^` as well? This is common in npm ranges